### PR TITLE
Fixing ITK-SNAP permission issue

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes issue #743

## Proposed Changes
  - disabled caching because there seems to be some issue going on in Azure regarding this
  - Permissions are now handled per-file instead of directory-wide (which didn't seem to work during install-time)
  - minor sanity check added in FE

### CODE REVIEW

File: src/applications/CMakeLists.txt [hunk 1]
```diff
@@ -430,9 +430,31 @@ FOREACH(subdir ${SUBDIRECTORIES})
...
         )
+
+        FOREACH(itkSnapLib_files ${FILES_SNAP})
+          INSTALL(
+            FILES ${itkSnapLib_files}
+            DESTINATION lib/${itkSnapLib}
+            PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
+          )
+        ENDFOREACH()
+
```
Comment (dboun) [inline src/applications/CMakeLists.txt]: Wouldn't it be easier to just make the one file (entry point) of itk-snap executable? Not everything has to be given the permission, especially directories which executable means a whole different thing.

Comment (dboun) [reply src/applications/CMakeLists.txt]: Oh, I just realized what you are doing. I guess this is easier; sorry

**Comment by dboun** [comment]
> Permissions are now handled per-file instead of directory-wide (which didn't seem to work during install-time)

_Minor note that might not matter anymore for this PR, but fyi:_ Directory permissions for linux don't batch-control what you can do with the files inside. They control stuff like whether you can ```cd``` into that directory, see the contents and create/remove files. In detail:

>    r - you have the ability to read the contents of the directory (ie do an ls)

>    w - you have the ability to write into the directory (ie create files and directories)

>    x - you have the ability to enter that directory (ie cd)

About your mac issue, note that whether you can remove a directory is controlled from whether you have write permission for its parent directory

_(Disclaimer: not 100% sure about all this info, it's pretty advanced :P)_

**Comment by sarthakpati** [comment]
Is this good to be merged?

**Pull request merged**